.github/workflows/gradle-cache.yml

```diff
@@ -0,0 +1,29 @@
+name: Update Gradle build cache
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '.idea/**'
+      - '.gitattributes'
+      - '.github/**.json'
+      - '.gitignore'
+      - '.gitmodules'
+      - '**.md'
+      - 'LICENSE'
+      - 'NOTICE'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+      - uses: gradle/gradle-build-action@v2
+        with:
+          gradle-home-cache-cleanup: true
+      - run: ./gradlew assembleDebug
```

Comment:
If so, why not just merge these 2 yml.

Reply:
On the main branch only `./gradlew assembleDebug` is run to populate the Gradle cache. On PR branches we want to run all (relevant) tests to make sure the changes didn't break any functionality.

The goal is to waste as little energy as possible. Also running the tests on `main` might not make much of a difference. It surely adds up, though. And as mentioned before, I think there's little benefit to running the tests on `main`.

I guess you could add some shell scripting or a custom Gradle task to decide which tasks to run depending on the branch. I didn't feel like looking into that, though. If this is bothering you and you want to take this on, please open another PR.